### PR TITLE
I15-1 (9.4pre) Replace NexusSlitsPosition with DeviceValueMultiPosition

### PR DESCRIPTION
**Pull request by Mark-Booth**
Now that a generic DeviceValueMultiPosition is working, there is no need
for the specialised NexusSlitsPosition class to exist.

Also add a ScannableValueTextProvider interface which can be used to get
GUI appropriate text out of a scannable when ScannableValueLabelProvider
needs it, rather than just using toString(), which might include
decorations such as class name which aren't appropriate to show in the
GUI.

### CODE REVIEW

File: org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/scan/DeviceValueMultiPosition.java [hunk 1]
```diff
@@ -18,49 +18,38 @@
...
-	private List<Double> values = new ArrayList<Double>() {{
-		add(0.0);
-		add(0.0);
-	}};
+	private Map<String, Double> values;
 
-	public double getX_gap() {
-		return values.get(0);
+	public DeviceValueMultiPosition() {
+		values = new LinkedHashMap<String, Double>();
```
Comment (gerring) [inline org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/scan/DeviceValueMultiPosition.java]: MINOR Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)

**Comment by gerring** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by gerring** [comment]
Did somebody review this? I allow committers to push in things but
hopefully they get someone to review. Reviews can be forced to occur but in
gerrit we also do not enforce it...

**Comment by Mark-Booth** [comment]
Sorry, pushed a change to gda-core which required it and would have failed without it. If I'd have realised you were back I would have asked you to do a quick review but teh changes don't seem controversial to me.

**Pull request merged**